sklearn/linear_model/_base.py

```diff
@@ -651,15 +714,16 @@ def _pre_fit(X, y, Xy, precompute, normalize, fit_intercept, copy,
             check_input=check_input, sample_weight=sample_weight)
     if sample_weight is not None:
         X, y = _rescale_data(X, y, sample_weight=sample_weight)
+
+    # FIXME: 'normalize' to be removed in v1.2.0
     if hasattr(precompute, '__array__'):
         if (fit_intercept and not np.allclose(X_offset, np.zeros(n_features))
-                or normalize and not np.allclose(X_scale,
-                                                 np.ones(n_features))):
-            warnings.warn(
-                "Gram matrix was provided but X was centered to fit "
-                "intercept, or X was normalized : recomputing Gram matrix.",
-                UserWarning
-            )
+                or normalize and not np.allclose(X_scale, np.ones(n_features)
+                                                 )):
+            warnings.warn("Gram matrix was provided but X was centered"
+                          " to fit intercept, "
```

Comment:
The previous warping was better black style.

Reply:
Note: although we have not yet migrated the full code base to the black code-style we agreed we should do in the future so it's better to aim for it in lines impacted by our PRs.